src/Functions/FunctionsLogical.cpp

```diff
@@ -650,8 +651,37 @@ ColumnPtr FunctionAnyArityLogical<Impl, Name>::executeImpl(
     ColumnsWithTypeAndName arguments = args;
 
     /// Special implementation for short-circuit arguments.
-    if (checkShortCircuitArguments(arguments) != -1)
-        return executeShortCircuit(arguments, result_type);
+    if constexpr (std::is_same_v<Name, NameAnd> || std::is_same_v<Name, NameOr>)
+    {
+        ColumnRawPtrs solid_args;
```

Comment:
Terms "solid" and "not_solid" args seem funny to the reader. Should that perhaps be "short_circuit_arg" / "non_short_circuit_arg"?

Also, high-level comments are lacking in l. 654 - 683, kindly add them.

Reply:
@lgbo-ustc I feel the naming is not good either, also maybe add some comment?

Reply:
update is in https://github.com/ClickHouse/ClickHouse/pull/74589